src/init/system_start_service.c

```diff
@@ -22,12 +25,20 @@ static const char *script_commands[] = {
 
 int system_start(void) {
 	const char *command;
+	char *argv[10];
+	int argc;
+	const struct cmd *cmd;
 
 	setup_tty(OPTION_STRING_GET(tty_dev));
 
 	array_foreach(command, script_commands, ARRAY_SIZE(script_commands)) {
-
-		system(command);
+		argc = cmdline_tokenize((char *)command, argv);
+		if (0 == strncmp(argv[0], "pthread", 7)) {
```

Comment:
After cmdline_tokenize argv[i] contains null terminated string and checked will be correct.
In the future we wants starting services as threads (not a task) for small devices but now I don't want design this code because it will require some harm dependences